### PR TITLE
WIP: typings update

### PR DESCRIPTION
**Pull request by sklinov**
Replace `any` with generics
Tried to use `enums` for `string` values

### CODE REVIEW

File: commandbar.d.ts [hunk 1]
```diff
@@ -1,99 +1,125 @@
...
-// Programmatic commands
-export type LinkCommandTemplate = {
-    type: 'link';
-    value: string;
-    operation: 'self' | 'blank' | 'router';
-};
+    // CommandBar.boot
+    export type CommandBarBootArgs = {
+      id: string;
+    } & Record<string, string | undefined>; // Any other arguments? Possible to list them or create an interface?
```
Comment (jamesaevans) [inline commandbar.d.ts]: Under the hood, `boot` passes its argument to `addContext`, so we can use `CommanadBarContext` here.

File: commandbar.d.ts [hunk 2]
```diff
@@ -1,99 +1,125 @@
...
+    export type CommandBarArgument =
+      | ContextArgument
+      | ProvidedArgument
+      | DependentArgument
+      | FunctionArgument
+      | SetArgument;
+
+    export type CommandBarCommandConfig = {
+      name: string;
+      text: string; // React.ReactNode type possible?
```
Comment (jamesaevans) [inline commandbar.d.ts]: `React.ReactNode` isn't possible yet, but we will probably add support for it soon.

File: commandbar.d.ts (hunk 1 above)
Comment (sklinov) [reply commandbar.d.ts]: @jamesaevans Fixed that

File: commandbar.d.ts (hunk 2 above)
Comment (sklinov) [reply commandbar.d.ts]: @jamesaevans Removed that comment

**Review by jamesaevans — APPROVED** [review]

**Pull request merged**